to/number_of_substrings_with_only_1s.py

```diff
@@ -0,0 +1,52 @@
+class TreeNode(object):
+    def __init__(self, val, idx_in_str):
+        self.val = val
+        self.idx_in_str = idx_in_str
+        self.last_idx_in_str = len(val) + idx_in_str - 1
+
+    def get_val(self):
+        return self.val
+
+    def get_idx(self):
+        return self.idx_in_str
+
+    def get_last_idx(self):
+        return self.last_idx_in_str
+
+
+def check_s_all_1s(s):
+    return s == '1' * len(s)
+
+
+def check_dict_has_key(k, d):
+    return k in d.keys()
+
+
+class Solution:
+    '''
+    substring to a tree(breadth first search)
+    '''
+    def numSub(self, s: str) -> int:
+        result = 0
+        my_queue = []  # for BFS
+        my_map = {}  # records whether certain substring handled before
+
+        for idx, val in enumerate(s):  # starts with only one char
```

Comment:
O(n)
https://stackoverflow.com/questions/31127594/complexity-of-enumerate/31127924